app/controllers/posts_controller.rb

```diff
@@ -20,7 +20,11 @@ def create
 
     respond_to do |format|
       format.html do
-        redirect_to user_post_path(@post.user.id, @post.id) if @post.save
+        if @post.save
+          redirect_to user_post_path(@post.user.id, @post.id)
+        else
+          flash[:alert] = 'Error: post is not published'
+        end
```

Comment:
- [ ] Kindly add flash message for successful posts

Reply:
Done✔